packages/aws-cdk-lib/aws-eks/lib/cluster.ts

```
@@ -2044,10 +2046,37 @@ export class Cluster extends ClusterBase {
    */
   public get eksPodIdentityAgent(): IAddon | undefined {
     if (!this._eksPodIdentityAgent) {
-      this._eksPodIdentityAgent = new Addon(this, 'EksPodIdentityAgentAddon', {
-        cluster: this,
-        addonName: 'eks-pod-identity-agent',
-      });
+      // check if stack already has a pod identity agent
+      const existing_addons = this.stack.node.findAll().filter(child =>
+        child instanceof Addon &&
+        // have to find the child, as Addon.addonName is a token
+        (child.node.tryFindChild('Resource') as CfnAddon)?.addonName === 'eks-pod-identity-agent' &&
+        (child.node.tryFindChild('Resource') as CfnAddon)?.clusterName === this.clusterName,
+      ) as Addon[];
+      if (existing_addons.length > 1) {
```

While its against recommendation, there can be multiple pod identity agents running for a given eks cluster for usecases such as testing/migration or multi-tenant clusters. Do we want to throw the error here explicitly or just choose the first one in the list like below?

---

I believe as an addon you can only have one instance of it, as the error from issue #32580 states: `Error message: "eks-pod-identity-agent already exists in stack"`

However I do see other installation methods as explained in the pod identity [README](https://github.com/aws/eks-pod-identity-agent/blob/main/README.md), so it may be possible to still install multiple outside of the scope of addons.

---

Yeah to my knowledge, Addons are just preconfigured helm charts, so I could see how you could install multiple instance of the pod identity chart. But that doesn't cause an instant error like multiple Addons does. I'm trying to solve that validation error. If the deployment error from multiple misconfigured pod identity charts occurs, I don't think we could check and prevent that.